Fix table alignment and padding.

SteveL-MSFT opened the pull request:
## PR Summary

The original change #5568 to remove extra padding didn't take into account alignment.

Fix logic to accommodate left, center, and right alignment in table for only right side padding and added test.
Removed over active debug assert when formatting was necessary.

Marked `Set-Date` tests as `-Pending` due to #6227 

Fix https://github.com/PowerShell/PowerShell/issues/6228
Fix https://github.com/PowerShell/PowerShell/issues/6234

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

SteveL-MSFT committed:
[feature]
The original change to remove extra padding didn't take into account alignment.
Fix logic to accommodate left, center, and right alignment in table and added test.

SteveL-MSFT committed:
[feature]
fix ImplicitRemoting test that validates formatting to use same instance due to formatting changes in this PR

vors commented:
Does it fix the #6234 as well?

sayujshajith commented:
This PR fixes a consistent crash that I was getting with any command that uses Format Table. 
The issue tracking that crash is here: https://github.com/PowerShell/PowerShell/issues/6234

SteveL-MSFT commented:
I think I figured out why the tests are failing.  Working on a fix.

SteveL-MSFT committed:
[feature]
only use loopback to same powershell instance for formatting test as the other tests implicitly expect Windows PowerShell
Update-FormatData in a new runspace so it doesn't pollute the global sessionstate causing problems later

SteveL-MSFT commented:
Linux/Mac failures due to https://github.com/PowerShell/PowerShell/issues/6227

SteveL-MSFT committed:
[feature]

mark Set-Date tests as failing due to #6227

SteveL-MSFT commented:
Marked Set-Date tests requiring `sudo` as Pending for now due to #6227

adityapatwardhan commented:
@SteveL-MSFT Can you add information about `Set-Date` change in the description.

SteveL-MSFT commented:
@adityapatwardhan done

daxian-dbw commented on src/System.Management.Automation/FormatAndOutput/common/TableWriter.cs:
```diff
@@ -497,8 +504,6 @@ private static string GenerateRowField(string val, int width, int alignment, Dis
                 return s;
             }
-            // we have to pad
-            System.Diagnostics.Debug.Assert(finalValueDisplayLength == width - 1, "padding is not correct");
```
[This is not a blocking comment]
@SteveL-MSFT I just noticed that in the `switch (alignment)` block below, `addPadding` is checked only when padding for `left alignment`, not when padding for `center alignment`. This is not in the same pattern as in the `switch (alignment)` block in `if (currentValueDisplayLength < width)`. Just want to bring it up so you can take a another look to see if it's intended.

SteveL-MSFT replied:
Good catch, let me check this.  I'll add an extra test for truncation.

SteveL-MSFT committed:
address Dongbo's feedback on truncation
